src/track/cue.h

```diff
@@ -104,7 +105,7 @@ class Cue : public QObject {
     mixxx::CueType m_type;
     mixxx::audio::FramePos m_startPosition;
     mixxx::audio::FramePos m_endPosition;
-    const int m_iHotCue;
+    mutable int m_iHotCue;
```

Comment:
Thanks for your review. I should have noted that this is still a raw version.
IIRC mutable here was just a quick fix for some build warning or error I didn't investigate when I noticed it because I wanted it to work somehow, and test it.

In general I'm not 100% sure if the naive solution of simply re-assigning the hotcue number is the way to go, but I suspect the only reason the cue number was const before was that no one thought of the reassign/move use case.

Reply:
Oh yeah, I didn't even notice that. const members are also an anti feature. Simplfy removing the specifier should make it work as expected. Then the only change you need to make is to ensure that you actually get a non-const `Cue` object when look for it in `swapHotcues`.

Reply:
> In general I'm not 100% sure if the naive solution of simply re-assigning the hotcue number is the way to go, 

I think it should be fine. Though my main concern is that it makes it easier to accidentally make two cues with the same index. An alternative solution could be to have a 
`Cue::swap(&Cue other)` method that does the swap internally without the need for a public `setHotCue` setter.

Reply:
> `Cue::swap(&Cue other)` method that does the swap internally

Sounds good, I'll try that.
Though I'd like to restrict it to hotcues for now, `Cue::swapHotcue(Cue& other)`, I don't see the need for swapping other (potentially inaccesseible, firstSound) cues (intro, outro, temp loop, main cue).